include/enums.h

```diff
@@ -5983,50 +5984,73 @@ enum FoldStateFlags {
 };
 
 typedef enum FoldType {
-    FOLD_TYPE_NONE                    = 0x0,
-    FOLD_TYPE_1                       = 0x1,
-    FOLD_TYPE_2                       = 0x2,
-    FOLD_TYPE_3                       = 0x3,
-    FOLD_TYPE_4                       = 0x4,
-    FOLD_TYPE_5                       = 0x5,
-    FOLD_TYPE_6                       = 0x6,
-    FOLD_TYPE_7                       = 0x7,
-    FOLD_TYPE_8                       = 0x8,    // color overlay?
+    FOLD_UPD_CLEAR                    = 0x0,    // or FOLD_UPD_INIT?
+    FOLD_TYPE_1                       = 0x1,    // unused?
+    FOLD_TYPE_2                       = 0x2,    // unused?
+    FOLD_TYPE_3                       = 0x3,    // after goomba 'sticker' ambush in kmr_09 unfurls
+    FOLD_UPD_WAVY                     = 0x4,    // Kolorado when injured and Sushie when underwater (* note: Sushie fold rendering is bugged and only occurs *before* going underwater)
+    FOLD_UPD_SET_ANIM                 = 0x5,
+    FOLD_UPD_SET_COLOR                = 0x6,    // modulate color (args: R, G, B)
+    FOLD_UPD_SET_ALPHA                = 0x7,    // modulate alpha (args: A)
+    FOLD_UPD_SET_TINT                 = 0x8,    // modulate color+alpha (args: R, G, B, A)
     FOLD_TYPE_9                       = 0x9,
     FOLD_TYPE_A                       = 0xA,
-    FOLD_TYPE_B                       = 0xB,
-    FOLD_TYPE_C                       = 0xC,
-    FOLD_TYPE_D                       = 0xD,    // noisy hologram -- used for ghostly star spirits and merlar
+    FOLD_UPD_COLOR_BUF_SET_B          = 0xB,
+    FOLD_UPD_COLOR_BUF_SET_C          = 0xC,
+    FOLD_UPD_HOLOGRAM                 = 0xD,    // ghostly star spirits and merlar (args: ???, staticAmt, ???, alphaAmt)
     FOLD_TYPE_E                       = 0xE,
     FOLD_TYPE_F                       = 0xF,
-    FOLD_TYPE_10                      = 0x10,
-    FOLD_TYPE_11                      = 0x11,
+    FOLD_TYPE_10                      = 0x10,   // unused?
+    FOLD_UPD_ALLOC_COLOR_BUF          = 0x11,   // args: count
 } FoldType;
 
+typedef enum FoldAnim {
+    FOLD_ANIM_SHOCK                   = 0x00, // used for Goombaria and Goompapa when Kammy drops the hammer block
+    FOLD_ANIM_SHIVER                  = 0x01, // used when Goombaria gives Mario a kiss
```

Comment:
I would think this is also used when hitting a hammer block that is stronger than the current hammer level

Reply:
Yes, you can see this on line 1554 of 77480.c:
`func_802DDFF8(playerStatus->anim, FOLD_UPD_SET_ANIM, FOLD_ANIM_SHIVER, 1, 1, 0, 0);`